File: Guide.Booking.Server/DTOs/BookingDto.cs
namespace Backend.DTOs
{
    public class BookingDto
    {
        public int Id { get; set; }
        public string CustomerName { get; set; } = string.Empty;
        public string CustomerEmail { get; set; } = string.Empty;
        public TourPackageDto TourPackage { get; set; } = new();
        public DateTime TourDate { get; set; }
        public int NumberOfPeople { get; set; }
        public List<ExtraDto> Extras { get; set; } = new();
        public List<TicketCategoryDto> TicketCategories { get; set; } = new();
        public decimal TotalPrice { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}

File: Guide.Booking.Client/src/types/booking.dto.ts
import type { TourPackageDto, ExtraDto, TicketCategoryDto } from './tour.dto';

export interface BookingDto {
  id: number;
  customerName: string;
  customerEmail: string;
  tourPackage: TourPackageDto;
  tourDate: string;
  numberOfPeople: number;
  totalPrice: number;
  extras: ExtraDto[];
  ticketCategories: TicketCategoryDto[];
  createdAt: string;
}

export interface BookingFormDto {
  customerName: string;
  customerEmail: string;
  tourPackageId: number;
  tourDate: string;
  numberOfPeople: number;
  extraIds?: number[];
  ticketCategories?: TicketCategoryDto[];
}

export interface BookingFilterDto {
  fromDate?: string;
  toDate?: string;
  tourPackageId?: number;
  minGroupSize?: number;
  maxGroupSize?: number;
}
